packages/portal/src/server-middleware/api/matomo/page-hits.js

```diff
@@ -0,0 +1,119 @@
+import axios from 'axios';
+import flatten from 'lodash/flatten.js';
+import http from 'http';
+import https from 'https';
+import { createClient } from 'redis';
+
+import { errorHandler } from '../index.js';
+
+const CACHE_KEY_PREFIX = '@europeana:matomo:page-hits';
+
+let matomoAxiosInstance;
+let redisClient;
+
+const pageHitsFromMatomo = async(pageUrl) => {
+  const matomoResponse = await matomoAxiosInstance.request({
+    params: {
+      pageUrl
+    }
+  });
+
+  const dayStats = flatten(Object.values(matomoResponse.data || {}))
+    .filter((stats) => stats.url === pageUrl);
+
+  const dayHits = dayStats.map((stats) => stats['nb_hits'] || 0);
+
+  return sum(dayHits);
+};
+
+const sum = (vals) => vals.reduce((memo, val) => memo + (val || 0), 0);
+
+const cacheKey = (url) => {
+  return `${CACHE_KEY_PREFIX}:${url}`;
```

Comment:
with URLs starting with `https://...` the `:` splits the cache key into a sub directory. It should probably be escaped.